apps/console/src/features/roles/components/edit-role/edit-role-common/permissions-list.tsx

```diff
@@ -45,36 +45,37 @@ interface PermissionsListPropsInterface extends  IdentifiableComponentInterface
      * Callback to handle API resource removal.
      */
     onChangeScopes: (apiResource: APIResourceInterface, scopes: ScopeInterface[]) => void;
+    /**
+     * Whether the Autocomplete field has an error or not.
+     */
+    hasError?: boolean;
+    /**
+     * Error message for the Autocomplete field.
+     */
+    errorMessage?: string;
 }
 
-export const PermissionsList: FunctionComponent<PermissionsListPropsInterface> = 
+export const PermissionsList: FunctionComponent<PermissionsListPropsInterface> =
     (props: PermissionsListPropsInterface): ReactElement => {
 
         const {
             apiResource,
             initialSelectedPermissions,
             selectedPermissions,
             onChangeScopes,
+            hasError,
+            errorMessage,
             [ "data-componentid" ]: componentId
         } = props;
 
         const { t } = useTranslation();
 
         const [ activeOption, setActiveOption ] = useState<ScopeInterface>(undefined);
-        const [ removedPermissions, setRemovedPermissions ] = useState<ScopeInterface[]>([]);
+        const [ hasErrorState, setHasErrorState ] = useState<boolean>(hasError);
```

Comment:
Any specific reason to define another state variable for a value that's retrieved from props? can't we directly use the prop value in respective places?

Reply:
Addressed: 16ae921f4144efa91f6b4f93e7bb21922243052b